src/mame/irem/olibochu.cpp

```diff
@@ -503,91 +510,128 @@ static INPUT_PORTS_START( punchkid )
 {
-	// basic machine hardware
-	Z80(config, m_maincpu, 18.432_MHz_XTAL / 6);
+	Z80(config, m_maincpu, MASTER_CLOCK / 3 / 2);
 	m_maincpu->set_addrmap(AS_PROGRAM, &olibochu_state::main_map);
-	TIMER(config, "scantimer").configure_scanline(FUNC(olibochu_state::scanline), "screen", 0, 1);
 
-	Z80(config, m_audiocpu, 18.432_MHz_XTAL / 6);
-	m_audiocpu->set_addrmap(AS_PROGRAM, &olibochu_state::sound_map);
-	m_audiocpu->set_periodic_int(FUNC(olibochu_state::irq0_line_hold), attotime::from_hz(120));
+	OUTPUT_LATCH(config, m_outlatch);
+	// Q0-6: see TODO
+	m_outlatch->bit_handler<7>().set(FUNC(olibochu_state::flip_screen_set));
+
+	// video
+	SCREEN(config, m_screen, SCREEN_TYPE_RASTER);
+	m_screen->set_raw(MASTER_CLOCK / 3, HTOTAL, HBEND, HBSTART, VTOTAL, VBEND, VBSTART);
+	m_screen->set_screen_update(FUNC(olibochu_state::screen_update));
+	m_screen->set_palette(m_palette);
 
-	// video hardware
-	screen_device &screen(SCREEN(config, "screen", SCREEN_TYPE_RASTER));
-	screen.set_refresh_hz(60);
-	screen.set_vblank_time(ATTOSECONDS_IN_USEC(0));
-	screen.set_size(32*8, 32*8);
-	screen.set_visarea(0*8, 32*8-1, 1*8, 31*8-1);
-	screen.set_screen_update(FUNC(olibochu_state::screen_update));
-	screen.set_palette(m_palette);
+	TIMER(config, m_timer).configure_scanline(FUNC(olibochu_state::scanline), "screen", 0, 1);
 
 	GFXDECODE(config, m_gfxdecode, m_palette, gfx_olibochu);
-	PALETTE(config, m_palette, FUNC(olibochu_state::palette), 256*2, 32);
+	PALETTE(config, m_palette, FUNC(olibochu_state::palette), 256 * 2, 32);
+
+	// audio
+	Z80(config, m_audiocpu, MASTER_CLOCK / 3 / 2);
+	m_audiocpu->set_addrmap(AS_PROGRAM, &olibochu_state::sound_map);
 
-	// sound hardware
 	SPEAKER(config, "mono").front_center();
 
 	GENERIC_LATCH_8(config, m_soundlatch[0]);
 	GENERIC_LATCH_8(config, m_soundlatch[1]);
 
-	AY8910(config, "aysnd", 18.432_MHz_XTAL / 12).add_route(ALL_OUTPUTS, "mono", 0.5);
+	AY8910(config, m_ay8910, MASTER_CLOCK / 3 / 2 / 2).add_route(ALL_OUTPUTS, "mono", 0.5);
 
-	HC55516(config, m_cvsd, 0).add_route(ALL_OUTPUTS, "mono", 0.5);
-	clock_device &cvsd_clock(CLOCK(config, "cvsd_clock", 16000));
-	cvsd_clock.signal_handler().set(FUNC(olibochu_state::cvsd_tick));
-	cvsd_clock.signal_handler().append(m_cvsd, FUNC(hc55516_device::mclock_w));
+	HC55516(config, m_cvsd, 0).add_route(ALL_OUTPUTS, "mono", 0.5); // see TODO
+	CLOCK(config, m_cvsd_clock, 16000); // see TODO
+	m_cvsd_clock->signal_handler().set(FUNC(olibochu_state::cvsd_tick));
+	m_cvsd_clock->signal_handler().append(m_cvsd, FUNC(hc55516_device::mclock_w));
 }
 
 
+/*************************************
+ *
+ *  ROM definitions
+ *
+ *************************************/
+/*
+Sound M-47C-A:
+
+                Z80
+             2114   OBC17
+             2114   OBC18
+  OBC15
+  OBC16
+
+  HC3-55536    8910
 
-/***************************************************************************
-    ROM Definitions
-***************************************************************************/
 
+CPU M-47A-A:
```

Comment:
Like I said in another PR: If the notes are at the top of the driver, you don't have to more game notes to the ROM defs, they're fine in the comment block at the top.